internal/ingress/controller/store/store.go

```diff
@@ -617,6 +617,11 @@ func New(checkOCSP bool,
 	return store
 }
 
+// CreatesCatchAll returns whether or not an ingress creates a catch-all backend
+func CreatesCatchAll(spec extensions.IngressSpec) bool {
+	return spec.Backend != nil && len(spec.Rules) == 0
+}
```

Comment:
What about implementing the whole logic here - including `disableCatchAll` flag check here.

Reply:
agreed but I'd change the function name.